src/main/java/seedu/trackbeau/logic/commands/customer/EditCustomerCommand.java

```diff
@@ -94,11 +94,18 @@ public CommandResult execute(Model model) throws CommandException {
 
         Customer customerToEdit = lastShownList.get(index.getZeroBased());
         Customer editedCustomer = createEditedCustomer(customerToEdit, editCustomerDescriptor);
-
-        if (!customerToEdit.isSameItem(editedCustomer) && model.hasCustomer(editedCustomer)) {
+        if (customerToEdit.isDifferentItemFromOriginal(editedCustomer) && model.hasCustomer(editedCustomer)) {
```

Comment:
I think the code might break here because hasCustomer(editedCustomer) uses isSameItem to check for existence.
If you just change email, isDifferentItemFromOriginal(editedCustomer) will return true and hasCustomer() will return true also since phone is still the same. That would throw an error but changing email should be allowed. 
Not sure will happen or not, can verify with a test case.
May can just stick with isSameItem and just check name & phone number.

Reply:
yea the logic gets a lot more complicated than I thought
isSameItem doesn't really work to prevent edits when e.g. only one of the identity fields has changed
there is a need to change hasCustomer to follow another logic if not users cannot edit identity fields of the same person even if there isn't clash with another customer